mysql-test/mysql-test-run.pl

```diff
@@ -2845,9 +2849,23 @@ ()
            <$bindir/lib/plugin/*.so>,             # bintar
            <$bindir/lib/plugin/*.dll>)
       {
-        my $pname=basename($_);
+        my $pname= basename($_);
         set_plugin_var($pname);
-        $plugindir=dirname($_) unless $plugindir;
+        $plugindir= dirname($_) unless $plugindir;
+      }
+
+      #
+      # Note: libmariadb3 is first, as in Debian, we
+      #       separate server plugins from client plugins.
+      for (<$bindir/lib/*/libmariadb3/plugin/*.so>,
+           <$bindir/lib64/mysql/plugin/*.so>,
+           <$bindir/lib/mysql/plugin/*.so>,
+           <$bindir/lib64/mariadb/plugin/*.so>,
+           <$bindir/lib/mariadb/plugin/*.so>,
+           <$bindir/lib/plugin/*.so>,             # bintar
+           <$bindir/lib/plugin/*.dll>)
+      {
+        $client_plugindir= dirname($_) unless $client_plugindir;
```

Comment:
may be `last` ? you don't do anything per plugin inside the loop

or, better, scan client-only dirs. Like:.
```perl
  for (<$bindir/lib/*/libmariadb3/plugin>) {
    $client_plugindir= $_ if <$_/*.so>;
  }
  $client_plugindir= $plugindir unless $client_plgindir;
```

note that neither solution works correctly if there's more than one libmariadb version installed. 
perhaps the only correct approach is to parse `mysql --help` to get the correct plugin-dir from there.

Reply:
This is @cvicentiu's commit, maybe he want's to amend?